bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/i18n/LocaleProvider.java

```diff
@@ -25,4 +28,10 @@
      */
     Locale getLocale();
 
+    /**
+     * Provides access to the location of the installation
+     *
+     * @return location of the current installation or null if the location is not set
+     */
+    PointType getLocation();
```

Comment:
> Why should a configuration be necessarily be separated from functionality?
￼
There is IMHO not a "it should" or "it must".

But if you need a location I would like an LocationProvider instance, if I need a locale I would like an LocationProvider intance and if I need to translate some stuff I would like an `I18nProvider`.

So, it is not about that the implementation could be done in one class but I don't need an implementation of a transaltion mechanism just because I would like to know the location or locale of a system.

IMHO that is the intention of interfaces to define a subset of stuff that should be combined and not to generate an interface that defines all the stuff that could ever be used or implemented by someone.